docs/proposals/suggestion.md

```diff
@@ -0,0 +1,408 @@
+# Suggestion CRD Design Proposal
+
+Table of Contents
+=================
+
+   * [Suggestion CRD Design Proposal](#suggestion-crd-design-proposal)
+   * [Table of Contents](#table-of-contents)
+      * [Background](#background)
+      * [Goals](#goals)
+      * [Non-Goals](#non-goals)
+      * [Design](#design)
+         * [Kubernetes API](#kubernetes-api)
+         * [GRPC API](#grpc-api)
+         * [Workflow](#workflow)
+            * [Example](#example)
+
+Created by [gh-md-toc](https://github.com/ekalinin/github-markdown-toc)
+
+## Background
+
+## Goals
+
+## Non-Goals
+
+## Design
+
+### Kubernetes API
+
+```go
+type SuggestionSpec struct {
+	//Name of the algorithm
+	AlgorithmName string `json:"algorithm_name"`
+
+	// Number of suggestions requested
+	Suggestions int `json:"suggestions"`
+
+	//Algorithm settings set by the user in the experiment config
+	AlgorithmSettings []AlgorithmSetting `json:"algorithm_settings,omitempty"`
+}
+
+type AlgorithmSetting struct {
+	Name  string `json:"name,omitempty"`
+	Value string `json:"value,omitempty"`
+}
+
+type SuggestionStatus struct {
+	// Suggestion results
```

Comment:
for some suggestion algorithm, some intermediate AlgorithmSettings which always updated after a new suggestion  need be stored here